### PR TITLE
New package: LimeSuite-20.10.0

### PR DESCRIPTION
**Pull request by classabbyamp**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

### CODE REVIEW

**Comment by classabbyamp** [comment]
### Current issues:
- [ ] aarch64 and aarch64-musl fail because LimeSuite's cmake adds `-march=native`, which doesn't work
- [ ] all the non-cross builds fail because LimeSuite's shlib isn't being found/resolved even though it's added to common/shlibs
  - this was previously resolved locally but has appeared again in the CI

**Comment by q66** [comment]
`-march=native` should be dropped on *every* architecture

**Comment by classabbyamp** [comment]
@q66, on x86, `-march=native` is never added because of logic in the cmake file. I've made sure it gets caught on arm though.

**Comment by q66** [comment]
that logic is still wrong since not just x86 and arm exists

**Comment by classabbyamp** [comment]
@q66 ok I'll just patch the -march line out of the cmake file instead

**Comment by q66** [comment]
this is not needed, your previous thing would've been fine, just make it not arch-conditional

there are no flags needed, the appropriate arch baseline defaults are builtin into the compiler

**Comment by classabbyamp** [comment]
for SIMD? This program does a lot of DSP/high-bandwidth data handling, so I would think enabling SIMD stuff would be wanted. See [upstream](https://github.com/myriadrf/LimeSuite/blob/master/CMakeLists.txt#L111-L141).

**Comment by q66** [comment]
x86_64 guarantees at least SSE2, so that's implied by the compiler, and anything newer is not available on all CPUs, so we don't want it in packages (if a binary SIGILLs on any computer Void can run on, that's bad news)

**Comment by classabbyamp** [comment]
So which option do you recommend be set, if any? `none native SSE3 SSE4.2 AVX AVX2`

**Comment by q66** [comment]
none

File: srcpkgs/LimeSuite/template [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Template file for 'LimeSuite'
+pkgname=LimeSuite
+version=20.10.0
+revision=1
+build_style=cmake
+build_helper=cmake-wxWidgets-gtk3
+configure_args="
+	-DENABLE_SIMD_FLAGS=none
```
Comment (Duncaen) [inline srcpkgs/LimeSuite/template:8]: we generally indent with a single space inside of quotes.

File: srcpkgs/LimeSuite/template [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+makedepends="gnuplot wxWidgets-gtk3-devel libusb-devel SoapySDR-devel libfreeglut-devel
+ fltk-devel glew-devel sqlite-devel libgomp-devel"
+depends="wget"
+short_desc="Driver and GUI for LimeSDR/LMS7002M-based SDR platforms"
+maintainer="classabbyamp <dev@kb6.ee>"
+license="Apache-2.0"
+homepage="https://myriadrf.org/projects/lime-suite/"
+distfiles="https://github.com/myriadrf/LimeSuite/archive/refs/tags/v${version}.tar.gz"
+checksum=f6d79dc67cb52a5aea839d1dc00e65f85367cb2c275d77f149833d32cf79b467
+shlib_provides="libLimeSuite.so.20.10-1"
```
Comment (abenson) [inline srcpkgs/LimeSuite/template:35]: I don't think this is needed

File: common/shlibs [hunk 3]
```diff
@@ -4062,3 +4062,5 @@ libomalloc-0.9.6.so singular-4.2.1_1
 libsingular_resources-4.2.1.so singular-4.2.1_1
 libm4ri-0.0.20200125.so m4ri-20200125_1
 libm4rie-0.0.20200125.so m4rie-20200125_1
+libkdumpfile.so.8 libkdumpfile-0.4.0_1
+libLimeSuite.so.20.10-1 LimeSuite-20.10.0_1
```
Comment (abenson) [inline common/shlibs:4066]: This probably isn't needed unless its used outside of `LimeSuite`

File: common/shlibs [hunk 4]
```diff
@@ -4062,3 +4062,5 @@ libomalloc-0.9.6.so singular-4.2.1_1
 libsingular_resources-4.2.1.so singular-4.2.1_1
 libm4ri-0.0.20200125.so m4ri-20200125_1
 libm4rie-0.0.20200125.so m4rie-20200125_1
+libkdumpfile.so.8 libkdumpfile-0.4.0_1
```
Comment (abenson) [inline common/shlibs:4065]: This should be its own commit for `libkdumpfile`

File: common/shlibs (hunk 3 above)
Comment (classabbyamp) [reply common/shlibs:4066]: it's needed for the SoapyLMS7 subpkg

File: common/shlibs (hunk 4 above)
Comment (classabbyamp) [reply common/shlibs:4065]: not even sure how that got in there, I have no memory of putting it there and it seem to build fine without...

File: srcpkgs/LimeSuite/template (hunk 2 above)
Comment (classabbyamp) [reply srcpkgs/LimeSuite/template]: I believe it was added because of this: https://github.com/void-linux/void-packages/runs/4196787171?check_suite_focus=true#step:9:156

File: srcpkgs/LimeSuite/template [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	pkg_install() {
+		vmove usr/include
+		vmove usr/lib/cmake
+		vmove usr/lib/pkgconfig
+		vmove "usr/lib/*.so"
+	}
+}
+
+LimeSuite-doc_package() {
+	pkg_install() {
```
Comment (abenson) [inline srcpkgs/LimeSuite/template:73]: `short_desc+=" - documentation"`

Comment (classabbyamp) [reply srcpkgs/LimeSuite/template:74]: added

**Comment by paper42** [comment]
please resolve the conflict in common/shlibs

**Comment by classabbyamp** [comment]
@paper42 done.
